controller.py

```diff
@@ -525,6 +561,27 @@ def dupe_check(candidate):
 
         return self.catalog_response(catalog, status_code)
 
+    @classmethod
+    def _update_library_authentication_url(
+            cls, library, new_authentication_url,
+            provided_shared_secret
+    ):
+        """Change a library's authentication URL, assuming the provided shared
+        secret gives the requester that permission.
+
+        :param library: A Library
+        :param new_authentication_url: A proposed new value for
+            Library.authentication_url
+        :param provided_shared_secret: Allegedly, the library's
+            shared secret.
+        """
+        if library.shared_secret != provided_shared_secret:
+            return AUTHENTICATION_FAILURE.detailed(
+                _("Provided shared secret is invalid")
+            )
```

Comment:
This can't happen the way we call `_update_library_authentication_url` currently; it's just for the sake of paranoia.